src/main/java/io/axoniq/axonserver/connector/impl/ControlChannelImpl.java

```diff
@@ -187,6 +190,12 @@ public synchronized void reconnect() {
 
     private void handleDisconnect(Throwable cause) {
         heartbeatMonitor.pause();
+        Status status = Status.fromThrowable(cause);
+        if (status == Status.UNAVAILABLE && channel.isReady()) {
```

Comment:
Nit: I'd might provide a `Supplier<Boolean>` to the constructor instead of the entire `AxonServerManagedChannel`. 
Mainly since the channel isn't used for anything else right now.

Then again, it conceptually doesn't sound weird that the channel impl. holds a channel (hence why I mark this as a nit).

Reply:
The `AxonServerManagedChannel` as already passed in the constructor. The only thing I did was to store it in a field, to be able to access it.

Reply:
That makes this nit obsolete if you ask me.